src/TextSerializationWriter.php

```diff
@@ -186,9 +186,15 @@ public function writeTimeValue(?string $key, ?Time $value): void
      */
     public function writeDateIntervalValue(?string $key, ?DateInterval $value): void
     {
-        if ($value) {
-            $valueString = "P{$value->y}Y{$value->y}M{$value->d}DT{$value->h}H{$value->i}M{$value->s}S";
-            $this->writeStringValue($key, $valueString);
+        if ($value !== null){
```

Comment:
instead of duplicating this logic all over the place, could we add a toString (or different name that makes more sense) in the abstractions?

Reply:
Please check https://github.com/microsoft/kiota-abstractions-php/pull/144